src/prommis/hydrogen_decrepitation/hydrogen_decrepitation_furnace.py

```diff
@@ -0,0 +1,1651 @@
+            within=PositiveReals,
+            initialize=1050,
+            units=pyunits.K,
+            bounds=(950.13, 1173.15),
+            doc="Temperature of insulation material 1 under steady operating conditions",
+        )
+
+        self.thickness_insulation_material1 = Var(
+            self.flowsheet().config.time,
+            within=PositiveReals,
+            initialize=0.12,
+            units=pyunits.m,
+            bounds=(0, None),
+            doc="Thickness of insulation material 1",
+        )
+
+        self.relative_thickness_ratio_insulation_material1 = Var(
+            self.flowsheet().config.time,
+            within=PositiveReals,
+            initialize=1,
+            units=pyunits.dimensionless,
+            bounds=(0, None),
+            doc="Relative thickness ratio of insulation material 1",
+        )
+
+        self.thickness_metal_material1 = Var(
+            self.flowsheet().config.time,
+            within=PositiveReals,
+            initialize=3 * 1e-3,
+            units=pyunits.m,
+            bounds=(0, None),
+            doc="Thickness of metal material 1",
+        )
+
+        self.relative_thickness_ratio_metal_material1 = Var(
+            self.flowsheet().config.time,
+            within=PositiveReals,
+            initialize=0.05,
+            units=pyunits.dimensionless,
+            bounds=(0, None),
+            doc="Relative thickness ratio of metal material 1",
+        )
+
+        self.thickness_insulation_material2 = Var(
+            self.flowsheet().config.time,
+            within=PositiveReals,
+            initialize=0.15,
+            units=pyunits.m,
+            bounds=(0, None),
+            doc="Thickness of insulation material 2",
+        )
+
+        self.relative_thickness_ratio_insulation_material2 = Var(
+            self.flowsheet().config.time,
+            within=PositiveReals,
+            initialize=1,
+            units=pyunits.dimensionless,
+            bounds=(0, None),
+            doc="Relative thickness ratio of insulation material 2",
+        )
+
+        self.thickness_metal_material2 = Var(
+            self.flowsheet().config.time,
+            within=PositiveReals,
+            initialize=4 * 1e-3,
+            units=pyunits.m,
+            bounds=(0, None),
+            doc="Thickness of metal material 2",
+        )
+
+        self.relative_thickness_ratio_metal_material2 = Var(
+            self.flowsheet().config.time,
+            within=PositiveReals,
+            initialize=0.01,
+            units=pyunits.dimensionless,
+            bounds=(0, None),
+            doc="Relative thickness ratio of metal material 2",
+        )
+
+        self.total_heat_duty = Var(
+            self.flowsheet().config.time,
+            within=PositiveReals,
+            initialize=3500,
+            units=pyunits.W,
+            bounds=(0, None),
+            doc="Total heat duty required for decrepitation",
+        )
```

Comment:
I do not like Pyomo domain restrictions beyond the distinction between integer and noninteger. If you try to change bounds, Pyomo silently truncates the bounds to remain within the domain.